cdap-client/src/main/java/co/cask/cdap/client/config/ConnectionConfig.java

```diff
@@ -58,6 +59,7 @@ private static String tryResolveAddress(String addressString) {
   private final int port;
   private final boolean sslEnabled;
   private final Id.Namespace namespace;
+  private String username;
```

Comment:
Can it be `final`? Meaning not set through setter method but have to be passed in through the constructor. Adding mutation field turn this class from immutable class to mutable one, making it not thread safe.